modules/rntuple.mjs

```diff
@@ -686,13 +715,21 @@ class RNTupleDescriptorBuilder {
 
     // Example Of Deserializing Page Content
     deserializePage(blob, columnDescriptor) {
-        const {
+        const originalColtype = columnDescriptor.coltype,
+        {
             blob: processedBlob,
             coltype
-        } = recontructUnsplitBuffer(blob, columnDescriptor),
-            byteSize = getTypeByteSize(coltype),
-            reader = new RBufferReader(processedBlob),
-            values = [];
+        } = recontructUnsplitBuffer(blob, columnDescriptor);
+        
+        // Handle split index types
+        if (originalColtype === ENTupleColumnType.kSplitIndex32 || originalColtype=== ENTupleColumnType.kSplitIndex64) {
+            const { blob: decodedArray } = DecodeDeltaIndex(processedBlob, coltype);
+            return decodedArray;  
```

Comment:
For uniformity, instead of returning the array here you can instead reassign `processedBlob` to it and let the function proceed as normal.
This may come in handy if later we want to change anything about the processing of a page